File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.py
# 다 더한 값에서 2가지 값을 빼서 100이 되도록 한다.

heightList=[]
x,y = 0,0
# 9 개의 값을 받는다.
for _ in range(9):
    height = int(input())
    heightList.append(height)

for i in range(9):
    for j in range(i+1,9):
        if sum(heightList) - (heightList[i]+heightList[j]) == 100:
            x = heightList[i]
            y = heightList[j]

heightList.remove(x)
heightList.remove(y)

heightList.sort()
print('\n'.join(map(str,heightList)))

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> dw;
vector<int> ans;
int sum = 0;
int N = 9;
int mx = 100;

void FindDw(int cnt, int res){
    if(res==100 && cnt==7){
        sort(ans.begin(),ans.end());
        for(int i=0;i<7;i++){
            cout<<ans[i]<<"\n";
        }
        exit(0);
    }
    
    for(int i=cnt;i<dw.size();i++){
        ans.push_back(dw[i]);
        FindDw(cnt+1,res+dw[i]);
        ans.pop_back();
    }
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
    for(int i=0;i<N;i++){
        int height;
        cin>>height;
        dw.push_back(height);
    }
    FindDw(0,0);
}
